examples/api/test/material/dialog/dialog.0_test.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/dialog/dialog.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Show Dialog', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Scaffold(
+          body: example.MyApp(),
+        ),
+      ),
+    );
+
+    expect(find.byType(FlutterLogo), findsNothing);
+
+    await tester.tap(find.text('Show Dialog'));
+    await tester.pumpAndSettle();
+    expect(find.byType(FlutterLogo), findsOneWidget);
+
+    await tester.tap(find.text('Close'));
+    await tester.pumpAndSettle();
+    expect(find.byType(FlutterLogo), findsNothing);
+  });
+
+  testWidgets('Show Dialog.fullscreen', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Scaffold(
+          body: example.MyApp(),
+        ),
+      ),
+    );
+
+    expect(find.byType(FlutterLogo), findsNothing);
+
+    await tester.tap(find.text('Show Dialog.fullscreen'));
+    await tester.pumpAndSettle();
+    expect(find.byType(FlutterLogo), findsOneWidget);
```

Comment:
not completely sure how these tests work, but would it make sense to have a different widget to test here instead of the same `FlutterLogo`.